sherpa/models/tests/test_model.py

```diff
@@ -1199,8 +1242,8 @@ def test_cache_not_used_in_fit():
     res = fit.fit(cache=False)
     assert mdl.c0.val == pytest.approx(2)
     assert len(mdl._cache) == 0
```

Comment:
Would it make sense for this to be

```suggestion
        assert mdl.cache == 0
```

(and similarly line 1238 checks `mdl.cache` rather than `len(mdl.cache)`)? The idea being that we try not to use "internal" stuff where possible so we can ensure the user API is useful / makes sense. Now, saying that, our tests are full of access to internals (sometimes it is necessary, sometimes it is code that could be updated but we haven't got a need for changing it, and sometimes it's in between). Here I've got myself confused in what the code is now doing, so I'm asking!

Reply:
[well, after fixing the extra indentation I added...]

Reply:
Here, we want to check that `fit.fit(cache=False)` does not use the cache. The best way to check that it's not use it to check that nothing has been inserted into the dictionary, which is what this line does.

`assert mdl.cache == 0` should actually fail, because the size of cache is > 0 (the default is 5), but even if `cache > 0`, the dict that holds the cached values can be empty because the cache just has not been filled yet.

Reply:
Ta. I don't think we need to add an API to check the "used" size of the cache as that just feels like TMI ;-)